src/Core/functions.php

```diff
@@ -311,10 +311,13 @@ function deprecationWarning(string $message, int $stackFrame = 1): void
 
         static $errors = [];
         $checksum = md5($message);
-        if (isset($errors[$checksum])) {
+        $duplicate = (bool)Configure::read('Error.enableDuplicatedDeprecations', false);
+        if (isset($errors[$checksum]) && !$duplicate) {
             return;
         }
-        $errors[$checksum] = true;
+        if (!$duplicate) {
+            $errors[$checksum] = true;
+        }
```

Comment:
this causes to not keep `checksum` if user want to show all deprecations

Reply:
also this preventing helps to use `$this->expectDeprecation()` after using `TestSuite::deprecated()`

but if user had 2 duplicate deprecation + `$this->expectDeprecation()`

he should enable duplicated deprecation